WalletWasabi/WabiSabi/Crypto/CredentialIssuer.cs

```diff
@@ -66,10 +74,9 @@ public class CredentialIssuer
 
 		// Keeps track of the used serial numbers. This is part of
 		// the double-spending prevention mechanism.
-		private HashSet<GroupElement> SerialNumbers { get; } = new HashSet<GroupElement>();
+		private ConcurrentDictionary<GroupElement, bool> SerialNumbers { get; } = new();
```

Comment:
Use simple collections with lock.

Reply:
the whole point of this PR is to reduce contention due to holding shared locks...

Reply:
actually thinking about it more, for this specific collection a regular collection would actually better, and result in less contention, so long as the lock isn't held while the crypto code is running.

but the Arena and Round collections should still be a concurrent dictionary so as to not block readers and avoid carrying variables out of critical regions